eng/enforce-issue.ps1

```diff
@@ -0,0 +1,42 @@
+[CmdletBinding()]
+param (
+	[Parameter(Mandatory = $True)]
+	$PullRequestId
+)
+
+Write-Host "Pull request ID: $PullRequestId"
+
+$prDetail = Invoke-WebRequest `
+	-UseBasicParsing `
+	-Uri "https://api.github.com/repos/dotnet/arcade-services/pulls/$PullRequestId" `
+| ConvertFrom-Json
+
+if ($prDetail.draft) {
+	Write-Host "Draft PR does not have to have GitHub issue specified. Check passed."
+	exit 0
+}
+
+$hasIssue = $prDetail.body -Match "github\.com/dotnet/(.+)/issues/(\d+)"
+if (-not $hasIssue) {
+	Write-Error "Link to the corresponding GitHub issue is missing in the PR description. Check failed."
```

Comment:
These need to be AzDO style errors so they actually show up. Otherwise it's going to be really frustrating to figure out why your build failed.